frontend/dialob-composer-semantic/src/components/TranslationDialog.js

```diff
@@ -185,65 +206,121 @@ const TranslationDialog = (props) => {
     return {translations, metadata};
   }
 
-  const getValueSetTranslations = () => {
-    if (props.valueSets) {
-      let translations = {};
-      let metadata = {key : {}};
-      props.valueSets.forEach((vs, k) => {
-        if (vs.get('entries')) {
-          vs.get('entries').forEach((e, k) => {
-            const key = `v:${vs.get('id')}:${k}:${e.get('id')}`;
-            translations[key] = e.get('label').toJS();
-            metadata.key[key] = {description: 'Valueset entry'};
-            return true;
-          });
-        }
-        return true;
-      });
-      return {translations, metadata};
-    } else {
-      return {translations: {}, metadata: {}};
+  const getAllTranslations = () => {
+    let translations = {};
+    let metadata = {key : {}};
+    let globalValueSetsTranslations = {};
+    let globalValueSetsMetadata = {key : {}};
+    let valueSets = props.valueSets.toJS();
+    let allGlobalValueSets = null;
+    if(props.globalValueSets){
+      allGlobalValueSets = props.globalValueSets.toJS();
+    }
+
+    function visitItem(item, pageId) {
+      if (item.label) {
+        const key = `i:${item.id}:l`;
+        translations[key] = item.label;
+        metadata.key[key] = {description: 'Item label', richText: item.type === 'note', pageId: pageId};
+      }
+      if (item.description) {
+        const key = `i:${item.id}:d`;
+        translations[key] = item.description;
+        metadata.key[key] = {description: 'Item description', richText: true, pageId: pageId};
+      }
+      if (item.validations) {
+        item.validations.forEach((val, idx) => {
+          const key = `i:${item.id}:v:${idx}`;
+          translations[key] = val.message;
+          metadata.key[key] = {description: 'Validation', pageId: pageId};
+          return true;
+        });
+      }
+
+      // Checking for valueSetId
+      if(item.valueSetId){
+        valueSets.forEach((valueSet) => {
+          if(valueSet.id === item.valueSetId && valueSet.entries){
+            if(!allGlobalValueSets?.some(globalValueSet => globalValueSet.valueSetId === item.valueSetId)){
+              valueSet.entries.forEach((entry, index) => {
+                const key = `v:${valueSet.id}:${index}:${entry.id}`;
+                translations[key] = entry.label;
+                metadata.key[key] = {description: 'Valueset entry', pageId: pageId};
+              })
+            }else{
```

Comment:
If I understand this correctly now, it would much more sense to export global valuesets outside of this loop as these are not necessarily connected to any item. Export only non-global valuesets with items.

It would make sense to refactor this code into more readable structure, have functions that perform specific tasks etc.